### ISSUE TITLE
Feature suggestion: Show title on mouseover on overflowing nav items

### ISSUE THREAD

**Issue by wilhelmer** [squidfunk/mkdocs-material#3785]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I want to suggest an idea and checked that ...

- [X] ... to my best knowledge, my idea wouldn't break something for other users
- [X] ... the documentation does not mention anything about my idea
- [X] ... there are no open or closed issues that are related to my idea

### Description

If a nav item has a very long title, the title is truncated with ellipsis, which is good to prevent the nav from scrolling horizontally. (This may be Insiders only, not sure.)

However, it would be nice if users could see the full title on mouseover, like this:

<img width="304" alt="2022-03-31_15-58-52" src="https://user-images.githubusercontent.com/2234833/161085866-12bfc110-8692-456a-8bee-c0bcae118778.png">

Basically, I can see two solutions for this:

1) Add a `title` attribute to **all** nav items. This can easily be done by modifying the `nav-item` partial. However, this creates a lot of redundancy, especially on large doc projects with huge navs.
2) Add a `title` attribute to overflowing nav items only. This can be done using JS. See code below. (Yes, it's jQuery! Heresy! But to get an idea.)

```js
// Add title to overflowing nav links
$(".md-ellipsis").each(function () {
    if (this.offsetWidth < this.scrollWidth) {
        $(this).attr("title", $(this).text().trim());
    }
});
```

### Use Cases

See description.

**Comment by squidfunk** [comment]
Thanks for suggesting! As a general note, I'm working on better hover behavior for elements with titles (including abbreviations specifically), and fixing this issue will be a part of it. It might take some time, but the logic is the same, so it wouldn't make sense to solve this specifically.

In the mean time, the following script should mitigate the issue (adapted from your code example to remove jQuery):

``` js
document$.subscribe(() => {
  document.querySelectorAll(".md-ellipsis").forEach(el => {
    if (this.offsetWidth < this.scrollWidth) {
      el.setAttribute("title", el.innerText)
    }
  })
})
```

Note that will only work in Insiders, since it makes use of the refactoring of the navigation structure to support icons.

**Comment by wilhelmer** [comment]
> Note that will only work in Insiders, since it makes use of the refactoring of the navigation structure to support icons.

Off-topic, but .. 😮 you can use icons in the nav? Is that documented somewhere? Couldn't find it.

**Comment by squidfunk** [comment]
@wilhelmer jup, they're called [page icons](https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-icon), since they are set per page. if you click the link and look to the left, you'll see the navigation making use of it 😊 This was added in __insiders-4.5.0__.

**Comment by wilhelmer** [comment]
This is awesome. Please consider linking to it from the "Icons, Emojis" page. It's not prominent enough in search, probably because it's a h3 section.

**Comment by squidfunk** [comment]
Makes sense, we could add it under the usage guide.

**Comment by wilhelmer** [comment]
One last thing – your code sample isn't working, must be `el` instead of `this`: `el.offsetWidth < el.scrollWidth`.

**Comment by wilhelmer** [comment]
Also, it won't work if the nav is collapsed at page load, because `offsetWidth` and `scollWidth` of all collapsed entries is 0 then.

It's a little trickier than I thought ...

**Comment by squidfunk** [comment]
Here's a video of the new ellipsis behavior that is landing next week in Insiders:

https://user-images.githubusercontent.com/932156/167295571-ccc635ca-996a-4c45-b717-14664f7abcd6.mp4

All elements that have ellipsis will now show a small tooltip when they overflow. The video shows that this is also the case for the header title that wraps on scroll. Additionally, content elements that have a `title` attribute can be opted into this new behavior as well by enabling the new `content.tooltips` flag:

<img width="172" alt="Bildschirmfoto 2022-05-08 um 14 15 32" src="https://user-images.githubusercontent.com/932156/167295687-8517459e-c615-4394-a4bd-db524fcd80e4.png">

**Comment by squidfunk** [comment]
Addendum: of course, it will also apply to the table of contents:

<img width="295" alt="Bildschirmfoto 2022-05-08 um 14 22 38" src="https://user-images.githubusercontent.com/932156/167295939-7b95e954-4a0d-4500-9ccb-2ad8a216481d.png">

**Comment by squidfunk** [comment]
__Insiders 4.15.0__ was just released and comes with [improved support for tooltips](https://squidfunk.github.io/mkdocs-material/reference/tooltips/#improved-tooltips) :tada: This release also includes the behavior mentioned above, automatically adding a tooltip to overflowing navigation titles.
